ksql-engine/src/main/java/io/confluent/ksql/ddl/commands/DropSourceCommand.java

```diff
@@ -76,60 +75,44 @@ public DdlCommandResult run(MetaStore metaStore, boolean isValidatePhase) {
         new DropTopicCommand(dataSource.getKsqlTopic().getTopicName());
     metaStore.deleteSource(sourceName);
     dropTopicCommand.run(metaStore, isValidatePhase);
-
     deleteTopicIfNeeded(dataSource, isValidatePhase);
-
     return new DdlCommandResult(true, "Source " + sourceName + " was dropped. "
-                                      + (deleteTopic ? "Topic '"
-                                                    + dataSource.getKsqlTopic().getTopicName()
-                                                    + "' was marked for deletion. Actual deletion "
-                                                    + "and removal from brokers may take some time "
-                                                    + "to complete." : ""));
+        + (deleteTopic ? "Topic '"
+        + dataSource.getKsqlTopic().getTopicName()
+        + "' was marked for deletion. Actual deletion "
+        + "and removal from brokers may take some time "
+        + "to complete." : ""));
   }
 
   private void deleteTopicIfNeeded(StructuredDataSource dataSource, boolean isValidatePhase) {
     if (!isValidatePhase && deleteTopic) {
-
-      executeWithRetries(new Callable<Void>() {
-        @Override
-        public Void call() throws Exception {
+      try {
+        ExecutorWithRetries.execute((Supplier<KafkaFuture<Void>>) () -> {
           kafkaTopicClient.deleteTopics(
-              Collections.singletonList(dataSource.getKsqlTopic().getKafkaTopicName()));
+              Collections.singletonList(
+                  dataSource.getKsqlTopic().getKafkaTopicName()));
           return null;
-        }
-      }, "Could not delete the corresponding kafka topic: "
-           + dataSource.getKsqlTopic().getKafkaTopicName());
-
+        });
+      } catch (Exception e) {
+        e.printStackTrace();
+      }
       if (dataSource.getKsqlTopic().getKsqlTopicSerDe().getSerDe()
           == DataSource.DataSourceSerDe.AVRO) {
-        executeWithRetries(new Callable<Void>() {
-          @Override
-          public Void call() throws Exception {
-            schemaRegistryClient
-                .deleteSubject(sourceName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
+        try {
+          ExecutorWithRetries.execute((Supplier<KafkaFuture<Void>>) () -> {
+            try {
+              schemaRegistryClient
+                  .deleteSubject(sourceName + KsqlConstants.SCHEMA_REGISTRY_VALUE_SUFFIX);
+            } catch (IOException | RestClientException e) {
+              e.printStackTrace();
```

Comment:
I know you copied this code, but I think it would be much better to instantiate a logger for this class and then log this exception as `LOG.error("Failed to delete subject from schema registry: {}", sourceName, e);`.

Reply:
Don't log and swallow the exception!  This kind of negates the point in executing it with retries ;) Personally, I'd go for:

```
try {
   execute(...);
} catch (final RuntimeException e) {
   throw e;
} catch (Exception e) {
   throw new RuntimeException(e);
}
```